src/content/docs/en/guides/middleware.mdx

```diff
@@ -0,0 +1,176 @@
+---
+title: Middleware (Experimental)
+description: Learn how to enable experimental middleware support in Astro.
+i18nReady: false
+---
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+**Middleware** is enabled in Astro behind an experimental flag. Middleware allows you to intercept requests and responses and inject behaviors dynamically every time a page or endpoint is about to be rendered. This also allows you to set and share request-specific information across endpoints and pages by mutating a `locals` object that is available in all Astro components and API endpoints. Middleware is available in both SSG and SSR Astro projects.
+
+:::caution
+Middleware is an experimental Astro feature introduced in v2.4. This API is subject to change before it is marked as stable.
+:::
+
+## Enabling Assets in your Project
+
+Enabling middleware should not cause any breaking changes in your project, but you may encounter errors or bugs as you use experimental middleware features.
+
+Please refer to the [middleware RFC](https://github.com/withastro/roadmap/pull/555) for more information.
+
+
+### Update Astro config
+
+To enable built-in asset support, add the following lines to your `astro.config.mjs` configuration file:
+
+```js title="astro.config.mjs" ins={4-6}
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  experimental: {
+   middleware: true
+  }
+});
+```
+
+## Basic Usage
+
+1. Create `src/middleware.js|ts` (Alternatively, you can create `src/middleware/index.js|ts`.)
+
+2. Inside this file, export an [`onRequest()`](#onrequest) function that modifies a [`locals`](#locals) object. This must not be a default export.
+
+```js title="src/middleware.js"
+export function onRequest ({ locals, request }, resolve) {
+	// locals = {}
```

Comment:
You can't set the value of `locals` here. You can add properties, `locals.foo = 'bar'`